diesel/src/migrations/migration.rs

```diff
@@ -65,10 +60,27 @@ impl Migration for SqlFileMigration {
     }
 }
 
-fn run_sql_from_file(conn: &SimpleConnection, path: &Path) -> Result<(), RunMigrationsError> {
+
+pub fn programmatic_migration_from(path: PathBuf) -> Result<String, MigrationError> {
+    if valid_sql_migration_directory(&path) {
+        let version = try!(version_from_path(&path));
+        let up = try!(sql_from_file(&path.join("up.sql")));
+        let down = try!(sql_from_file(&path.join("down.sql")));
+        Ok(format!("ProgrammaticMigration{{ version:\"{}\", up:\"{}\", down: \"{}\"}}",version.trim(),up.trim(),down.trim()))
```

Comment:
Code shouldn't get constructed by string munging. Can you move this to an annotation in codegen?